src/com/facebook/buck/features/apple/project/ProjectGenerator.java

```diff
@@ -930,18 +930,6 @@ private PBXNativeTarget generateAppleBundleTarget(
                       filteredRules.add(node);
                     }
                   });
-          TargetNodes.castArg(node, PrebuiltAppleFrameworkDescriptionArg.class)
-              .ifPresent(
-                  prebuiltFramework -> {
-                    // Technically (see Apple Tech Notes 2435), static frameworks are lies. In case
-                    // a static framework is used, they can escape the incorrect project generation
-                    // by marking its preferred linkage static (what does preferred linkage even
-                    // mean for a prebuilt thing? none of this makes sense anyways).
-                    if (prebuiltFramework.getConstructorArg().getPreferredLinkage()
-                        != NativeLinkableGroup.Linkage.STATIC) {
-                      filteredRules.add(node);
-                    }
-                  });
```

Comment:
According to tech notes https://developer.apple.com/library/archive/technotes/tn2435/_index.html, "apps cannot embed it (static framework) in the app bundle". we already excluded static framework here, why do you want to remove the code?

Reply:
I may not be 100% correct in my explanation since the code was originally from @rmaz, but from what I gather we cannot filter the rule here because we need to propagate the dependencies during the project generation

 The static frameworks are being filtered only on the copy files build phase. (Line 3427)

Reply:
How about non-static, NativeLinkableGroup.Linkage.SHARED for example? should it be added to filteredRules?
I have tested with and without this change, the test passes in both cases. That means either this change is not needed, or the test does not cover the change.

Reply:
I think the code here was made redundant by the other changes, but we can revert this part if you think that's better. All my manual tests work in both cases.

Reply:
Please revert. Thank you

Reply:
Reverted, but it looks like a CI test failed for unrelated reasons, can you force a rerun?

Reply:
That's unrelated. I'll fix it later.